src/parser.rs

```diff
@@ -295,39 +295,17 @@ impl Library {
             _ => Err(parser.unexpected_element(elem)),
         })?;
 
-        let disguised = elem.attr_bool("disguised", false);
-        if disguised && (c_type == "GdkAtom" || c_type == "GIConv") {
-            // A 'disguised' structure is one where the c:type is a typedef that
-            // doesn't look like a pointer, but is internally: typedef struct _X *X;
-            //
-            // Currently g-ir-sanner is too eager to mark all typedef to non-complete
-            // types as disguised, so we limit this behaviour two most common cases.
-            // https://gitlab.gnome.org/GNOME/gobject-introspection/issues/101
-
-            let disguised_record_name = record_name.clone();
-            let disguised_c_type = c_type.clone();
-
-            record_name = format!("_{}", &record_name);
-            c_type = format!("_{}", &c_type);
-            let target_c_type = format!("{}*", c_type);
-
-            let tid = self.find_or_stub_type(ns_id, &record_name);
-            self.add_type(
-                ns_id,
-                &disguised_record_name,
-                Type::Alias(Alias {
-                    name: disguised_record_name.clone(),
-                    c_identifier: disguised_c_type,
-                    typ: tid,
-                    target_c_type,
-                    doc: None, //TODO: temporary
-                })
-            );
-        }
+        // A 'disguised' structure is one where the c:type is a typedef that
+        // doesn't look like a pointer, but is internally: typedef struct _X *X;
```

Comment:
What about using this as a comment for Record::disguised field?
Seems like a more appropriate place now.

Reply:
Good idea, it will be first comment in `library::Record` :wink: